tests/api.spec.ts

```diff
@@ -0,0 +1,168 @@
+        public readonly method: string,
+        public readonly headers: object,
+        public readonly url: string,
+        public readonly body?: object,
+    ) {}
+
+    public static readonly CreateUser = new Requests("POST", { "Content-Type": "application/json" }, "api/users", {
+        name: "aqa",
+        job: "aqa engeneer",
+    });
+    public static readonly UpdateUser = new Requests("PUT", { "Content-Type": "application/json" }, "api/users/2", {
+        name: "morpheus",
+        job: "zion resident",
+    });
+    public static readonly DeleteUser = new Requests("DELETE", { "Content-Type": "application/json" }, "api/users/2");
+    public static readonly LoginUser = new Requests("POST", { "Content-Type": "application/json" }, "api/login", {
+        email: "eve.holt@reqres.in",
+        password: "cityslicka",
+    });
+    public static readonly RegisterUser = new Requests("POST", { "Content-Type": "application/json" }, "api/register", {
+        email: "eve.holt@reqres.in",
+        password: "pistol",
+    });
+}
+
+describe("API test suite", () => {
+    const createUserData = Requests.CreateUser;
+    const updateUserData = Requests.UpdateUser;
+    const deleteUserData = Requests.DeleteUser;
+    const loginUserData = Requests.LoginUser;
+    const registerUserData = Requests.RegisterUser;
+
+    it(`status code check all users`, async () => {
+        try {
+            const response = await superagent.get("https://reqres.in/api/users");
+            console.log(response.body as resp);
+            expect(response.status).toBe(200);
+        } catch (err: any) {
+            expect(err.message).toBe("");
+        }
+    });
+    it(`status code check single user`, async () => {
+        try {
+            const response = await superagent.get("https://reqres.in/api/users").query({ page: 2, per_page: 1 });
+            console.log(response.body as resp);
+            expect(response.status).toBe(200);
+            expect(response.body.per_page).toBe(1);
+        } catch (err: any) {
+            expect(err.message).toBe("");
+        }
+    });
+
+    it.skip(`status code check single user not found`, async () => {
+        try {
+            const response = await superagent.get("https://reqres.in/api/users/23");
+            expect(response.status).toBe(200);
+        } catch (err: any) {
+            expect(err.message).toBe("");
+        }
+    });
+
+    it(`create user`, async () => {
+        try {
+            const request = superagent
+                .get(`https://reqres.in/api/users`)
+                .send(createUserData.body)
+                .set(createUserData.headers);
+            request.method = createUserData.method;
+
+            const response = await request;
+            console.log(response.body);
+            expect(response.status).toBe(201);
+            expect(response.body.name).toEqual(expectedObj.name);
+        } catch (err: any) {
+            expect(err.message).toBe("");
```

Comment:
Please use something in toBe assertion, as for now you just log empty string. For instanse you could pass `"UNKNOWN ERROR"`